server/service/integration_mdm_dep_test.go

```diff
@@ -350,19 +351,26 @@ func (s *integrationMDMTestSuite) runDEPEnrollReleaseDeviceTest(t *testing.T, de
 	require.True(t, profileFleetCASeen)
 	require.True(t, profileFileVaultSeen)
 
+	// simulate fleetd being installed and the host being orbit-enrolled now
+	enrolledHost.OsqueryHostID = ptr.String(mdmDevice.UUID)
+	orbitKey := setOrbitEnrollment(t, enrolledHost, s.ds)
+	enrolledHost.OrbitNodeKey = &orbitKey
+
 	if enableReleaseManually {
 		// get the worker's pending job from the future, there should not be any
 		// because it needs to be released manually
 		pending, err := s.ds.GetQueuedJobs(ctx, 1, time.Now().UTC().Add(time.Minute))
 		require.NoError(t, err)
 		require.Empty(t, pending)
 	} else {
-		// there should be a Release Device pending job in the near future, expect
-		// it and schedule it to run now.
-		s.expectAndScheduleReleaseDeviceJob(t)
+		// there shouldn't be a Release Device pending job anymore
+		pending, err := s.ds.GetQueuedJobs(ctx, 1, time.Now().UTC().Add(time.Minute))
+		require.NoError(t, err)
+		require.Len(t, pending, 0)
 
-		// run the worker to process the DEP release
-		s.runWorker()
+		// call the /status endpoint to automatically release the host
+		var statusResp getOrbitSetupExperienceStatusResponse
+		s.DoJSON("POST", "/api/fleet/orbit/setup_experience/status", json.RawMessage(fmt.Sprintf(`{"orbit_node_key": %q}`, *enrolledHost.OrbitNodeKey)), http.StatusOK, &statusResp)
```

Comment:
Note that I plan to create another PR next week with more integration tests for this flow, so that it is tested with software to install/script to run and with the `force_release` bool, but I wanted to create the PR so that it can unblock the corresponding `fleetd`/Swift Dialog work.